python/cudf/dataframe/column.py

```diff
@@ -87,6 +87,10 @@ def __init__(self, data, mask=None, null_count=None):
         null_count : int; optional
             The number of null values in the mask.
         """
+        from .index import RangeIndex
+        # Convert range to RangeIndex
+        if isinstance(data, range):
+            data = RangeIndex(data)
         # Forces Column content to be contiguous
         if not data.is_contiguous():
```

Comment:
If `data` is a RangeIndex, and the `is_contiguous` method from the RangeIndex class is called, will `not data.is_contiguous()` always be False since RangeIndex.is_contiguous() always returns True?

Reply:
This is for Column initialization, not Index initialization, so there's no reason to be using a RangeIndex at all here.

Reply:
@beckernick If `data` is a `RangeIndex`, then it already is contiguous. If it is not a `RangeIndex`, we want to force whatever it is to be contiguous. I think what you said is true, and desired behavior.

@kkraus14 This code was failing when I added a `range(10)` as the first column of an empty `DataFrame`.  This was my fix. I'm not sure of its correctness but I am at 100% passing tests. There were lots of small issues to work through to get back to 100% test pass after we worked out the `Index` address fix. The whole day :D

Reply:
Removing this didn't affect test performance, thanks!